lua/weapons/cfc_slappers/shared.lua

```diff
@@ -0,0 +1,470 @@
+
+    -- Emit slap sound
+    self:GetOwner():EmitSound( table.Random( self.Sounds.Slap ), 80, math.random( 92, 108 ) )
+end
+
+function SWEP:SlapWorld( _, tr )
+    self:GetOwner():EmitSound( table.Random( self.Sounds.HitWorld ), 80, math.random( 92, 108 ) )
+    self:GetOwner():EmitSound( table.Random( self.Sounds.Slap ), 80, math.random( 92, 108 ) )
+    -- i just slapped a wall! that oughta hurt!
+    local damage = math.random( 1, 2 )
+    local dmginfo = DamageInfo()
+    dmginfo:SetDamageType( DMG_CLUB )
+    dmginfo:SetAttacker( self:GetOwner() )
+    dmginfo:SetInflictor( game.GetWorld() )
+    dmginfo:SetDamage( damage )
+    self:GetOwner():TakeDamageInfo( dmginfo )
+
+    -- Apply force to self
+    local origVel = self:GetOwner():GetVelocity()
+    local vec = ( tr.HitPos - tr.StartPos ):GetNormal()
+    local mul = GetConVar( "slappers_force" ):GetInt()
+    local slapVel = -vec * mul
+    local vel = slapVel * 0.6 + origVel --
+    self:GetOwner():SetLocalVelocity( vel )
+
+end
+
+function SWEP:SlapProp( ent, tr )
+    local vec = ( tr.HitPos - tr.StartPos ):GetNormal()
+    local emitSound = self.Sounds.HitWorld
+    local damage = math.random( 4, 6 )
+
+    if ent:GetClass() == "func_button" then
+        ent:Use( self:GetOwner(), self:GetOwner() ) -- Press button
+    elseif string.match( ent:GetClass(), "door" ) then
+        ent:Use( self:GetOwner(), self:GetOwner() ) -- Open door
+    elseif ent:Health() > 0 then
+        if ent:Health() <= damage then
+            ent:Fire( "Break" )
+        else
+            -- Damage props with health
+            local dmginfo = DamageInfo()
+            dmginfo:SetDamagePosition( tr.HitPos )
+            dmginfo:SetDamageType( DMG_CLUB )
+            dmginfo:SetAttacker( self:GetOwner() )
+            dmginfo:SetInflictor( self:GetOwner() )
+            dmginfo:SetDamage( damage )
+            ent:TakeDamageInfo( dmginfo )
+        end
+    end
+
+    -- Apply force to prop
+    local phys = ent:GetPhysicsObject()
+
+    if IsValid( phys ) then
+        emitSound = table.Random( self.Sounds.Slap )
+        phys:SetVelocity( phys:GetVelocity() + vec * GetConVar( "slappers_force" ):GetInt() * math.Clamp( 100 / phys:GetMass(), 0, 1 ) )
+    end
+
+    -- Emit slap sound
+    self:GetOwner():EmitSound( emitSound, 80, math.random( 92, 108 ) )
+end
+
+--[[
+	Third Person Slap Hack
+]]
+function SWEP:SlapAnimation()
+    -- Inform players of slap
+    if SERVER and not game.SinglePlayer() then
+        net.Start( "SlapAnimation" )
+        net.WriteEntity( self:GetOwner() )
+        net.Broadcast()
+    end
+
+    -- Temporarily change hold type so that we
+    -- can use the crowbar melee animation
+    self:SetWeaponHoldType( "melee" )
+    self:GetOwner():SetAnimation( PLAYER_ATTACK1 )
+
+    -- Change back to normal holdtype once we're done
+    timer.Simple( 0.3, function()
+        if IsValid( self ) then
+            self:SetWeaponHoldType( self.HoldType )
+        end
```

Comment:
Do we need to reset the animation too?
```suggestion
        if not IsValid( self ) then return end
        self:SetWeaponHoldType( self.HoldType )
```